lib/board.rb

```diff
@@ -1,2 +1,62 @@
 class Board
+  #1) how to populate gird with ships
+  #2) need to display board and update
+
+  attr_accessor :grid
+
+  def initialize(grid = Board.default_grid)
+    @grid = grid
+  end
+
+  def self.default_grid
+    Array.new(10) { Array.new(10)}
+  end
+
+  def [](pos)
+    row, col = pos
+    grid[row][col]
+  end
+
+  def over?
+    grid.flatten.none? { |pos| pos.nil? }
+  end
+
+  def populate_grid
+#how many ships do we populate with? 5
+#create random size
+#how to randomly place on grid, mark with :s
+#need help
+    #rand_ship_size = rand(1..grid.length)
+    #rand_ship_start = [rand(1..grid.length), rand(1..grid.length)
+    #5.times do |ship|
```

Comment:
You can rely on #place_random_ship and have simply : 

```ruby
5.times { place_random_ship }
```

Reply:
okay, so I created a help method 
def populate_grid 
    5.times { place_random_ship } 
end 

to automatically populate the grid, would I call this method during initialize? 

def initialize(grid = Board.default_grid)
    @grid = grid
    populate_grid 
end

Reply:
You can call `populate_grid` in the initialize of the board. But it could also be the responsibility of the `BattleshipGame` to trigger this action for example in `#play`:

```ruby
  def play
     board. populate_grid # "Place the pieces on the board"
     play_turn until game_over? "Start to play turns"
  end
```